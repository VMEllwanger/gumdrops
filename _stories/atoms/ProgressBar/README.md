## Props

The following props may be passed to configure the progress bar. All additional props are passed through to the root `<div>` element:

| name              | type      | description                                                          | default |
| ----------------- | --------- | -------------------------------------------------------------------- | ------- |
| className         | `String`  | Additional class to be added to the outermost element                |         |
| isSecondary       | `Boolean` | Indicate if bar color is secondary                                   | `false` |
| isStriped         | `Boolean` | Indicate if bar is striped                                           | `false` |
| size              | `String`  | Indicate the size of the bar. One of `xs`, `sm` or `lg`              |         |
| useProgressColors | `Boolean` | Indicate if the bar should change from red to green when progressing |         |
| value             | `number`  | A number (`0` to `100`) representing progress percent                |         |
