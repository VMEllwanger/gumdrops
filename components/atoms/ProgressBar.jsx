import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

const ProgressBar = ({
    className,
    isSecondary,
    isStriped,
    size,
    useProgressColors,
    value,
    ...rest
}) => {
    const baseClass = 'gds-progress-bar';
    const rootClass = cx(baseClass, className, {
        [`${baseClass}--${size}`]: !!size,
        [`${baseClass}--striped-primary`]: isStriped && !isSecondary,
        [`${baseClass}--striped-secondary`]: isStriped && isSecondary,
        [`${baseClass}--secondary`]: isSecondary && !isStriped,
        'gds-progress-bar--value-colors': useProgressColors
    });

    return <div className={rootClass} data-value={value} {...rest} />;
};

ProgressBar.propTypes = {
    className: PropTypes.string,
    isSecondary: PropTypes.bool,
    isStriped: PropTypes.bool,
    size: PropTypes.oneOf(['xs', 'sm', 'lg']),
    useProgressColors: PropTypes.bool,
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired
};

export default ProgressBar;
