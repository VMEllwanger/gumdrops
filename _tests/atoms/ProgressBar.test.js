/* globals shallow */
import React from 'react';
import ProgressBar from '../../components/atoms/ProgressBar';

const defaultProps = {
    isSecondary: false,
    isStriped: false,
    useProgressColors: false,
    size: 'sm',
    value: 50,
    className: 'foo-class'
};

describe('Expect <ProgressBar>', () => {
    it('to render', () => {
        const wrapper = shallow(<ProgressBar {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to renders as secondary', () => {
        const props = {
            ...defaultProps,
            isSecondary: true
        };
        const wrapper = shallow(<ProgressBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to renders as striped', () => {
        const props = {
            ...defaultProps,
            isStriped: true
        };
        const wrapper = shallow(<ProgressBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('to render colorful', () => {
        const props = {
            ...defaultProps,
            useProgressColors: true
        };
        const wrapper = shallow(<ProgressBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
