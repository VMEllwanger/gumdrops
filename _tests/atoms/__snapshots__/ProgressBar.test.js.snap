// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Expect <ProgressBar> to render 1`] = `
<div
  className="gds-progress-bar foo-class gds-progress-bar--sm"
  data-value={50}
/>
`;

exports[`Expect <ProgressBar> to render colorful 1`] = `
<div
  className="gds-progress-bar foo-class gds-progress-bar--sm gds-progress-bar--value-colors"
  data-value={50}
/>
`;

exports[`Expect <ProgressBar> to renders as secondary 1`] = `
<div
  className="gds-progress-bar foo-class gds-progress-bar--sm gds-progress-bar--secondary"
  data-value={50}
/>
`;

exports[`Expect <ProgressBar> to renders as striped 1`] = `
<div
  className="gds-progress-bar foo-class gds-progress-bar--sm gds-progress-bar--striped-primary"
  data-value={50}
/>
`;
